com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeReferenceVolume.cs

```diff
@@ -1163,75 +1093,93 @@ internal void Clear()
         }
 
         // Runtime API starts here
-        RegId AddBricks(List<Brick> bricks, ProbeBrickPool.DataLocation dataloc, ProbeBrickIndex.CellIndexUpdateInfo cellUpdateInfo, out List<Chunk> ch_list)
+        bool AddBricks(CellInfo cellInfo, ProbeBrickIndex.CellIndexUpdateInfo cellUpdateInfo)
         {
             Profiler.BeginSample("AddBricks");
 
+            var cell = cellInfo.cell;
+            var bricks = cell.bricks;
+
             // calculate the number of chunks necessary
-            int ch_size = m_Pool.GetChunkSize();
-            ch_list = new List<Chunk>((bricks.Count + ch_size - 1) / ch_size);
-            m_Pool.Allocate(ch_list.Capacity, ch_list);
-
-            // copy chunks into pool
-            m_TmpSrcChunks.Clear();
-            m_TmpSrcChunks.Capacity = ch_list.Count;
-            Chunk c;
-            c.x = 0;
-            c.y = 0;
-            c.z = 0;
-
-            // currently this code assumes that the texture width is a multiple of the allocation chunk size
-            for (int i = 0; i < ch_list.Count; i++)
+            int chunkSize = ProbeBrickPool.GetChunkSize();
+            int brickChunksCount = (bricks.Count + chunkSize - 1) / chunkSize;
+            cellInfo.chunkList.Clear();
+
+            // Try to allocate texture space
+            if (!m_Pool.Allocate(brickChunksCount, cellInfo.chunkList))
+                return false;
+
+            // In order not to pre-allocate for the worse case, we update the texture by smaller chunks with a preallocated DataLoc
+            int chunkIndex = 0;
+            while (chunkIndex < cellInfo.chunkList.Count)
```

Comment:
This looks correct, but is delicate, have you stress tested this a bit to see if all looks fine?

Reply:
I tested as much as I could but it could be interesting to test with bigger/more complex scenes as well. As it is, it's hard to force streaming activity in a relevant way :/